### PR TITLE
Enable tests on Stable versions

### PR DESCRIPTION
**Pull request by DanTup**
We disabled this in the past because of the increased time, however since we managed to make a build that didn't work for current Flutter stable, I think the value outweighs the cost. It did show up in the stable runs of [Dart Code's integration tests](https://dartcode.org/test-results/?master/c7a9044a73633c5281f0e4fa5d15d6c390978a49) but only after it was released.

Changes are:

- User stable Dart SDK
- Checkout flutter stable branch when using stable Dart SDK
- Skip the "main" bot for stable builds - this is mostly things that don't affect end users, like formatting, tuneup, which may legitimately fail (for ex. if the formatter changes and we update our source, it will fail to pass dartfmt check on stable until it catches up).

I expect some of the builds to fail against this PR because of the SDK constraints until https://github.com/flutter/devtools/pull/547 lands (if they don't, then this isn't working right). After landing that, I'll rebase this, and it should then go green.

@devoncarew @jacob314

### CODE REVIEW

**Comment by DanTup** [comment]
Less tests failed than I expected, because I overlooked that the Stable Dart SDK is v2.2 (all is good), it's only Flutter's Stable that fails. It does fail with the expected issue:

> Dart 2.1.2-dev.0.0.flutter-0a7dcf17eb is incompatible with your dependencies' SDK constraints. Please run "pub get" again.

We may want to consider whether we should run the non-Flutter bots using stable Flutter's version of the SDK, but I think as-is this covers the majority of cases.

I'll rebase this on master with the fix and all should be green.

**Comment by DanTup** [comment]
Tests are still red, but they're getting past all the activation/etc. and failing due to the goldens being different (Flutter changed slightly since then - similar to formatting - which we skip by skipping the main bot). We'll need to think about how to handle that, I'll come back to it.

**Comment by jacob314** [comment]
We should pass in whether the flutter version is stable or not as a Dart environment variable for the tests or better yet figure out how to determine it from the test runner by querying flutter to determine its release. The golden code can then append `_stable`  for the stable version of the golden files. It will mean twice as many goldens but updating the goldens for a release will still be a single command.

**Comment by jacob314** [comment]
lgtm

**Comment by DanTup** [comment]
Rebased this after landing #558, then pushed two new changesets:

- 70d3e45 - add stable versions of goldens
- 571cf37 - update Travis script to use the stable goldens for stable runs

Hopefully this will see Travis builds go green :-)

**Comment by DanTup** [comment]
@jacob314 please check you're happy with the two latest changests I added, then I believe this is good to go.

Well, except... this failed with this on stable:

```
05:07 +21 -1: test/inspector_controller_test.dart: inspector controller tests hotReload [E]
  TimeoutException after 0:04:00.000000: Test timed out after 4 minutes.
  package:test_api  Invoker._onRun.<fn>.<fn>.<fn>
```

My guess is that it was probably https://github.com/dart-lang/sdk/issues/33838 (we don't have verbose logging there to confirm) which was fixed, but hasn't made it to stable.

Therefore, we could hold off landing this until the nexts table release, or we could put some temporary code into the top of that test to just quit if it's the current stable version of Flutter (so it'll run only for the ones we know have the fix) and remove it after the next stable release.

WDYT?

**Comment by jacob314** [comment]
Yes I agree that is the flaky test issue. Just ignore and commit.

**Comment by DanTup** [comment]
I was more worried about making the builds generally flaky than this specific instance. I don't think it'd be a lot of effort to make this test skip for this current stable version of Flutter - I'm happy to implement it (and revert that once the next stable build goes live) if it sounds reasonable.

**Comment by jacob314** [comment]
If you want to skip the tests of stable that would be even better.

File: packages/devtools/test/inspector_controller_test.dart [hunk 1]
```diff
@@ -514,6 +514,14 @@ void main() async {
     });
 
     test('hotReload', () async {
+      if (flutterVersion == '1.2.1') {
```
Comment (jacob314) [inline packages/devtools/test/inspector_controller_test.dart]: Good idea using the exact flutter version to skip! That makes sure we don't lose the test.

**Comment by DanTup** [comment]
Landing this with a red build because we believe the failures are unrelated.

**Pull request merged**